closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -685,13 +685,8 @@ private void initProviders() {
                 // Provider with argument (provider name + optional argument).
                 providers.add(pNum - 1, providerName);
```

Comment:
SunSASL is not a FIPS provider, so we can not, by default, adding "RestrictedSecurity.NSS.140-2.jce.provider.5 = SunSASL" into java.security file. Clients need to make their own decision to add this SunSASL or not. We can update the FIPS document to mention this.

Reply:
Suppose a client includes either of these lines in `java.security`:
```
RestrictedSecurity.OpenJCEPlusFIPS.FIPS140-3.jce.provider.4 = SunSASL-NSS-FIPS
RestrictedSecurity.OpenJCEPlusFIPS.FIPS140-3.jce.provider.4 = com.sun.security.sasl.Provider
```
If I understand the intent of this change, either should result in
```
security.provider.4=SunSASL
```
being stored in the `Properties` object by `RestrictedSecurity.setProviders()`, but the code currently will not do that. Should it?

Reply:
It's not clear that you have answered my questions:
* Given a provider name with an argument (e.g. `-NSS_FIPS`) should it be removed or not when constructing the value of `security.provider.N`?
* If the provider name differs from the class name (like `com.sun.security.sasl.Provider`) should the value of `security.provider.N` be the provider name or the class name?

Comments in `java.security` discuss the different situations where the provider name vs. the class name are used. Did you test both?

Reply:
Given a provider name with an argument (e.g. -NSS_FIPS), we need to remove the argument when add it into variable `private final List<String> providersSimpleName;`. 

For creating the `security.provider.N` list, it will use the variable `private final List<String> providers`, which comes from the "`RestrictedSecurity.NSS.140-2.jce.provider.N`". So, if "`RestrictedSecurity.NSS.140-2.jce.provider.N=provider name`", then use provider name. If "`RestrictedSecurity.NSS.140-2.jce.provider.N=class name`", then use class name.

If the provider name differs from the class name (like com.sun.security.sasl.Provider), when creating the `security.provider.N` list, same as above first question, it will use the variable `private final List<String> providers`, which comes from the "`RestrictedSecurity.NSS.140-2.jce.provider.N`".

Reply:
I understand how `providers` and `providersSimpleName` are used. You still haven't answered my question. If this change is to have any value, a user will need to specify
```
RestrictedSecurity.some-profile.jce.provider.N = something
```
What should "_something_" be? Should it be `SunSASL` or `com.sun.security.sasl.Provider`?

[`java.security`](https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/openj9/src/java.base/share/conf/security/java.security#L75) suggests that `security.provider.N=SunSASL` is appropriate, so if "_something_" is `com.sun.security.sasl.Provider` we don't get the desired result.

Reply:
It should be `SunSASL`.  Example as follow:

`RestrictedSecurity.NSS.140-2.jce.provider.5 = SunSASL`